.github/workflows/release-preview.yml

```diff
@@ -0,0 +1,197 @@
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check if there are any changes to create a preview release for
+        id: changes
+        run: |
+          LATEST_PREVIEW=$(git tag --sort=-creatordate | grep -E '\.[0-9]{14}$' | head -n 1)
+          
+          DIFF=""
+          if [ -n "$LATEST_PREVIEW" ]; then
+            # using || true to always exit either with a diff or a success exit code to not fail the whole workflow
+            DIFF=$(git diff $LATEST_PREVIEW..5.x-dev --unified=0 | grep -vE "^\+\+\+|---" | grep "^[+-]" | grep -v "public const VERSION = '.*';" || true)
+          fi
+          
+          if [ -z "$DIFF" ]; then
+            echo "No changes in 5.x-dev since last preview version was created."
+            DO_RELEASE=0
+          else
+            DO_RELEASE=1
+          fi
+          
+          echo "do_release=$DO_RELEASE" >> $GITHUB_OUTPUT
+
+      - name: Determine new preview version number
+        if: steps.changes.outputs.do_release == '1'
+        id: version
+        run: |
+          OLD_VERSION=$(php -r "include_once 'core/Version.php'; echo \Piwik\Version::VERSION;")
+          NEW_VERSION=$(php -r "include_once 'core/Version.php'; \$v = new \Piwik\Version(); echo \$v->nextPreviewVersion(\Piwik\Version::VERSION);")
+          
+          if [ "$NEW_VERSION" == "" ]; then
+            HAS_NEW_VERSION=0
+          else
+            HAS_NEW_VERSION=1
+          fi
+          
+          echo "OLD_VERSION=$OLD_VERSION" >> $GITHUB_ENV
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+          
+          echo "has_new_version=$HAS_NEW_VERSION" >> $GITHUB_OUTPUT
+          echo "new_version=$NEW_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Update 5.x-preview branch to latest 5.x-dev
+        if: steps.changes.outputs.do_release == '1' && steps.version.outputs.has_new_version == '1'
+        run: |
+          git checkout -B 5.x-preview
+
+      - name: Update version file with new version
+        if: steps.changes.outputs.do_release == '1' && steps.version.outputs.has_new_version == '1'
+        run: |
+          sed -i "s/VERSION = '${OLD_VERSION}';/VERSION = '${NEW_VERSION}';/g" core/Version.php
+
+      - name: Commit version file changes
+        if: steps.changes.outputs.do_release == '1' && steps.version.outputs.has_new_version == '1'
+        run: |
+          git add core/Version.php
+          git commit -m "Update version to ${NEW_VERSION}"
+
+      - name: Push changes to 5.x-preview
+        if: steps.changes.outputs.do_release == '1' && steps.version.outputs.has_new_version == '1'
+        run: |
+          git push -f origin 5.x-preview
+
+  run_matomo_tests:
+    needs: [prepare_preview]
+    uses: ./.github/workflows/matomo-tests.yml
+    with:
+      is_preview: true
+
+  release_preview_version:
+    needs: [run_matomo_tests]
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
This now duplicates most of the release action code. Any reason for not reusing the release action we already have here?

Reply:
Personally I would vote for reusing the action, even if that would mean to adjust the other action so it also also works for releasing a preview on the new preview branch.

Reply:
Will adjust it here to use the action so that we only have one action producing releases.

Reply:
@sgiehl reused the release action, please re-review. Thanks!